slam_toolbox/src/slam_toolbox_common.cpp

```diff
@@ -446,8 +447,9 @@ bool SlamToolbox::shouldProcessScan(
 
   // check moved enough, within 10% for correction error
   const double sq_dist_to_last_accepted_pose = last_pose.SquaredDistance(pose);
+  const double head_to_last_accepted_pose = fmod(fabs(pose.GetHeading() - last_pose.GetHeading()), M_PI) / M_PI * 180;
 
-  if(sq_dist_to_last_accepted_pose < 0.8 * dist_thresh_sq || scan->header.seq < 5)
+  if((sq_dist_to_last_accepted_pose < 0.8 * dist_thresh_sq &&  head_to_last_accepted_pose < head_thresh)|| scan->header.seq < 5)
```

Comment:
Wouldn't this be or?

Reply:
According to my understanding, either minimum_travel_heading or minimum_travel_distance is exceeded, the scan should be processed. So, when return false, it should be "and".

Reply:
My issue is that we found that the param minimum_travel_heading didn't actually work like minimum_travel_distance. Our IMU is not accurate, so pure rotation is expected to be processed as well by function shouldProcessScan.